src/client/Microsoft.Identity.Client/Utils/Guard.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Identity.Client.Utils
+{
+    static class Guard
```

Comment:
```suggestion
    internal static class Guard
```

Reply:
Explicit is better than implicit...

https://peps.python.org/pep-0020/

CC @rayluo for the inspiration